components/brave_new_tab/resources/components/app.style.ts

```diff
@@ -0,0 +1,94 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { color, font } from '@brave/leo/tokens/css/variables'
+import { scoped, global } from '../lib/scoped_css'
+
+export const style = scoped.css`
+
+  .settings {
+    --leo-icon-size: 20px;
+
+    position: absolute;
+    inset-block-start: 4px;
+    inset-inline-end: 4px;
+    block-size: 20px;
+    inline-size: 20px;
+    opacity: 0.5;
+    color: #fff;
+    filter: drop-shadow(0px 1px 4px rgba(0, 0, 0, 0.60));
+
+    &:hover {
+      opacity: 0.7;
+      cursor: pointer;
+    }
+  }
+
+  main {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    min-height: 100vh;
+    padding-top: 40px;
+  }
+
+  .topsites-container {
+    min-height: 32px;
+  }
+
+  .searchbox-container {
+    flex: 1 1 auto;
+    margin: 16px 0;
+  }
+
+  .background-caption-container {
+    margin: 8px 0;
+  }
+
+  .widget-container {
+    min-height: 8px;
+  }
+
+`
+
+global.css`
+  @scope (${style.selector}) {
```

Comment:
The CSS defined here (via `global.css`) will "punch through" nested elements that have `data-css-scope`. We shouldn't do this much, but it's useful for defining app-wide standard tag styles and CSS classes.

Reply:
Hmmm, so you mean this is actually global CSS? What's the `@scope` for then?

In this case, I think we should either have a global `style.css` file or add this in to the `<style>` tag in `new_tab_page.html` - this feels like we're doing something more complicated than we need to

Reply:
The scope rule ensures that the styles apply to everything underneath this element but not above in scope. In this case, it doesn't matter (since it's the root element), but I think it's better to scope everything in general.

Would it be simpler to have styles spread among `html`/`css` and also `*.style.ts` files? It might seem odd when reading it at first to have these "punch-through" styles here, but it feels pretty natural to maintain.

Reply:
I think if we do want to keep these global styles here its probably better to remove the scope (it feels odd have global CSS which is then scoped).

This feels to me like its part of our CSS Reset for this page most of which is in `.html` file, but there's a little bit more here. Maybe we should move that here, or this there?

Reply:
As a general principle I think that all styles (even app-level) should be scoped, and it seems to work out in practice (e.g. in storybook the old styles no longer apply after HMR). This is another aspect that is not yet set-in-stone; let's see how it develops as we get more experience with scoped styles. 😄